src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -249,16 +250,32 @@ private function formatListForPresentation(CombinationListForEditing $combinatio
                 'isSelected' => false,
                 'name' => $combination->getCombinationName(),
                 'reference' => $combination->getReference(),
-                //@todo: don't forget image path when implemented in the query
                 'impactOnPrice' => (string) $combination->getImpactOnPrice(),
                 'quantity' => $combination->getQuantity(),
                 'isDefault' => $combination->isDefault(),
+                'imageUrl' => $this->formatImageUrl($combination->getRelativeImagePath()),
             ];
         }
 
         return $data;
     }
 
+    /**
+     * @param string|null $relativeImgPath
+     *
+     * @return string
+     */
+    private function formatImageUrl(?string $relativeImgPath): string
```

Comment:
If the query uses tha appropriate service (`prestashop.adapter.product.image.product_image_url_factory`) then the returned object will already contain the image URL So you can remove this method